nengo_loihi/hardware/snips/nengo_host.cc.template

```diff
@@ -70,48 +72,44 @@ class NengoHostProcess : public ConcurrentHostSnip {
 
     void run(std::atomic_bool& end_of_execution) override {
         // --- loop to transmit data from superhost to chip and vice versa
-        int step_counter = -1;
-
         while (!end_of_execution) {
-            // Wait until host sends number of run steps
-            while (!end_of_execution && step_counter <= 0) {
-                ssize_t n_read0 = recv(client_socket, buffer, sizeof(int32_t), MSG_DONTWAIT);
-                if (n_read0 == 4) {
-                    step_counter = buffer[0];
-                    if (step_counter > 0) {
-                        std::cout << "Host running " << step_counter << " steps" << std::endl;
-                    }
-                    break;
-                } else if (n_read0 > 0) {
-                    std::cout << "Host received unexpected number of bytes ("
-                              << n_read0 << ")" << std::endl;
-                    return;
-                } else {
-                    usleep(100);
-                }
-            }
-            if (end_of_execution) {
-                break;
-            } else if (step_counter <= 0) {
-                break;
-            }
-
-            // read messages from superhost socket
-            // we make sure BUFFER_SIZE is large enough for any message
-            // TODO: read in blocks of e.g 4096 bytes (small power of 2) for speed
-            size_t n_read = read(client_socket, buffer, BUFFER_SIZE * sizeof(int32_t));
-            if (n_read % sizeof(int32_t) != 0) {
-                std::cout << "Did not read full int32s from socket" << std::endl;
-                break;
-            }
+            // read packets from superhost socket
+            // first read is blocking, so we wait for packets
+#if DEBUG
+            std::cout << "Host: Preparing to read socket" << std::endl;
+#endif
+            ssize_t n_read = recv(client_socket, buffer, 4*PACKET_SIZE, 0);
```

Comment:
The packet size in bytes `4*PACKET_SIZE` can be converted to `PACKET_BYTE_SIZE` to have better clarity on why the packet size is multiplied by 4

Reply:
We could have `PACKET_ELEMENTS` for the size in `int32`s, and `PACKET_BYTES` for the size in bytes.